account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -291,17 +301,40 @@ def _update_statement_balance_values(
         statement_date_since,
         statement_date_until
     ):
-        """Update statement start or end balance from line data."""
+        """Update statement start or end balance from line data.
+
+        Statements retrieved and statements created might have different begin
+        and end dates. Suppose we retrieve data that covers three days, and has four
+        transactions, but the statement to be created is for only one day. like this:
+        statement_values: {
+            "balance_start": 25.00,
+            "balance_end_real": 150.00,
+        }
+        lines_data = [
+            {"date": "2020-02-01", "amount": 25.00},
+            {"date": "2020-02-02", "amount": 25.00},
+            {"date": "2020-02-02", "amount": 50.00},
+            {"date": "2020-02-03", "amount": 25.00},
+        ]
+        If we create a statement for only the day 2020-02-02, the first and
+        the last line should not belong to this statement. But at the start of
+        2020-02-02 the balance was 50.00. So the amounts for lesser dates should
+        be added to the balance_start. Similarly amounts from later dates should
+        be subtracted from balance_end_real. Which should be 125.00 for
+        2020-02-02.
```

Comment:
Okay, I get this... but:

- Let's say yesterday we had statement A
- Today, we have statement B, but during fetching of transactions for statement B, some transactions for end of yesterday still come through.
- We couple these now to statement B, but we add the items from yesterday to the starting balance

That's all fine and merry, but who then updates the end balance of yesterday? That has also shifted because of the new transactions being there.

Reply:
No, the extra transactions for statement A (yesterday) are NOT coupled to statement B. They are added to statement A, causing its end balance to be updated. After this statement B (today) will be done.

Because our last run (yesterday) was before the end of the day, each run (scheduled pull ) will retrieve first the transactions from yesterday, start of day. It will discard the transactions already imported with help of the unique_import_id. Then it will retrieve the transactions for today, as far as already available.